drivers/dma/dma_mcux_edma.c

```diff
@@ -24,13 +24,21 @@
 #include <zephyr/logging/log.h>
 #include <zephyr/irq.h>
 
+#ifdef CONFIG_DMA_MCUX_EDMA_LEGACY
 #define DT_DRV_COMPAT nxp_mcux_edma
+#elif CONFIG_DMA_MCUX_EDMA_V3
+#define DT_DRV_COMPAT nxp_mcux_edma_v3
+#endif
 
 LOG_MODULE_REGISTER(dma_mcux_edma, CONFIG_DMA_LOG_LEVEL);
 
 struct dma_mcux_edma_config {
 	DMA_Type *base;
-	DMAMUX_Type *dmamux_base;
+	DMAMUX_Type **dmamux_base;
+#ifdef CONFIG_DMA_MCUX_EDMA_V3
+	uint32_t channel_gap[2];
```

Comment:
does it make sense to support the channel gaps only for EDMA V3? it doesn't seem to be tied to the version of the EDMA but to a design of the SoC

Reply:
Just because currently in upstream, i don't see SoCs use the old version of EDMA has the gap

Reply:
I made it common for both